core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,738 @@
+                    container.remove_by_id(id.id);
+                }
+                Err(TransactionSchedulingError::UnschedulableConflicts) => {
+                    num_unschedulable += 1;
+                    // If configured, stop on first unschedulable transaction.
+                    if self.config.stop_on_unschedulable {
+                        // Push popped ID back into the queue.
+                        container.push_id_into_queue(id);
+                        break;
+                    } else {
+                        self.unschedulables.push(id);
+                    }
+                }
+                Ok(TransactionSchedulingInfo {
+                    thread_id,
+                    transaction,
+                    max_age,
+                    cost,
+                }) => {
+                    assert!(
+                        self.working_account_set.take_locks(&transaction),
+                        "locks must be available"
+                    );
+                    saturating_add_assign!(num_scheduled, 1);
+                    batches.transactions[thread_id].push(transaction);
+                    batches.ids[thread_id].push(id.id);
+                    batches.max_ages[thread_id].push(max_age);
+                    saturating_add_assign!(batches.total_cus[thread_id], cost);
+
+                    // If target batch size is reached, send all the batches
+                    if batches.ids[thread_id].len() >= self.config.target_transactions_per_batch {
+                        saturating_add_assign!(num_sent, self.send_batches(&mut batches,)?);
+                        self.working_account_set.clear();
+                    }
+
+                    // if the thread is at max_cu_per_thread, remove it from the schedulable threads
+                    // if there are no more schedulable threads, stop scheduling.
+                    if self.in_flight_tracker.cus_in_flight_per_thread()[thread_id]
+                        + batches.total_cus[thread_id]
+                        >= max_cu_per_thread
+                    {
+                        schedulable_threads.remove(thread_id);
+                        if schedulable_threads.is_empty() {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.working_account_set.clear();
+        self.send_batches(&mut batches)?;
+
+        // Push unschedulables back into the queue
+        for id in self.unschedulables.drain(..) {
+            container.push_id_into_queue(id);
+        }
+
+        Ok(SchedulingSummary {
+            num_scheduled,
+            num_unschedulable,
+            num_filtered_out,
+            filter_time_us: total_filter_time_us,
+        })
+    }
+
+    /// Receive completed batches of transactions without blocking.
+    /// Returns (num_transactions, num_retryable_transactions) on success.
+    fn receive_completed(
```

Comment:
This function is an exact copy of `PrioGraphScheduler::receive_completed`.
Duplicating to avoid refactors in a PR intended to be backported to 2.1.